### DISCUSSION TITLE
Tool vs Platform

### DISCUSSION THREAD

**Discussion by 5e-Cleric** [naturalcrit/homebrewery#3666] [General]
Discussions about different features of this website have for some time risen an important debate.

Some features, such as new markdown syntaxes, enhance the tool as it is, but other features like the upcoming Vault or a possible future with favorites, stride towards a full platform website. This is a topic of debate, because there are devs of the opinion that such features should exist separately from The Homebrewery.

I do not think that is the case and promote the tool as a place to edit, visualize, share and look for other brews, but one man's opinion is not the team's.

This is a place to have such conversations without cluttering other issues or the Gitter Chat.

**Comment by dbolack-ab** [comment]
I am unsure what are the platform features are vs tool features?

**Reply by 5e-Cleric** [reply]
In general, features that build on making documents would enhance the tool and features that are not related, and tend to go more towards users interacting between them would be on the side of platform.

For example, adding acutal user details to the userpage, like a user image, bio, and that. would be towards the platform. Same goes for stuff like adding comments or "likes" to brews.

Features like collab live editing, snippet widgets and adding themes does focus on building documents, and is talked in this discussion as "tool"

**Reply by Gazook89** [reply]
Some people want to make PDFs or links to share on places like Reddit or patreon. HB is just a tool. 

Some people want to follow authors directly, or build a following via likes and even comments, on HB. No longer just a tool, but a place where users connect with others (even if not communicating like with comments). 

If there is anything like “likes” that somehow is reflected in search results or feeds, it can be gamed and may be something that requires some level of moderation.

**Reply by dbolack-ab** [reply]
Some of this issue with likes depends on how far out the information goes. If likes are only shown to the who *liked* then it's functionally just a bookmarking system. If the information is shared back to the author, it's just a feedback system. If it's shared to the world at large, that's yet another thing.

**Reply by Gazook89** [reply]
Yeah, that's a good point and good to keep in mind.  Kind of like the difference between "Likes" and "Favorites", even though the difference between those two words is minimal (and in the context of the conversation there is plenty of cross-meaning).  

I personally am fine with bookmarking, your first scenario.

I personally think the second scenario is a little silly, and not terribly helpful for people to be seeking validation based on "likes" for their brews.  But there are probably worse ills in society, so whatever.

I hesitate to take number of "likes" and return it in the Vault results (which is like your third scenario). That to me is a level of 'promotion' of otherwise unmoderated content that I wouldn't want to get mixed up in.

**Reply by 5e-Cleric** [reply]
I would disagree, it is a way to categorize a brew depending on its quality, by being peer-reviewed. Like upvotes in reddit.

**Reply by Gazook89** [reply]
> Like upvotes in reddit.

This, to me, isn't the selling point you think it is.  Upvotes =/= quality.  Additionally, Reddit theoretically puts a lot of work (?) into curating a certain algorithm such that stuff gets cycled through, and the most upvoted stuff doesn't just sit at the top collecting more and more upvotes due to increased visibility.  If an HB user searches for "cat dragons" and only ever sees the most-liked brews on the first page of results, only those most-liked brews will ever accrue more likes and it will be self perpetuating.

**Reply by dbolack-ab** [reply]
When I saw likes desired as a concept for this project, I assumed ( based on the near non-existence of user accounts ) that liking a brew was more like a bookmark for the "liker" that also happened to tell the owner(s) of the document that someone liked it.

I suppose it was a weird conclusion to make, but given all of the other "missing/deliberately not created" social creations, it just read like an in-application bookmark.  I think that is the best way to treat it. Having some part of the data roll back to the brew might be nice, but I'm unsure how reliably we could couple/decouple that information. Similarly, I think it's a feature that would only work for users with logins as local storage could get purged.

**Reply by dbolack-ab** [reply]
As an adjunct to this one of my *working* assumptions extended from this thinking was that *eventually* users would be able to see favorited `meta:theme` brews on their theme pulldown without having to clone their own copy.

**Reply by 5e-Cleric** [reply]
that would be ideal, although the link functionality would be interesting too.

**Comment by calculuschild** [comment]
I don't think we have the manpower, security, infrastructure, or income needed for a whole "platform", There are plenty of other tools already in place for the social/sales/promotion/communication side of things that are going to be better than anything we can come up with. I just want the Homebrewery to be a cool tool for turning Markdown into documents and reading them.

Certain things like the Vault do straddle that line, but I think our intent is just to make it easier to share documents. Not any kind of social media / shop aspect to it. I would be especially nervous allowing users to message each other or comment on documents, etc.  That opens up a whole can of worms about moderation and another level of legal responsibility that I don't think we are equipped to handle on a volunteer basis as we are.

**Reply by 5e-Cleric** [reply]
thinking back, maybe vault should have been a separate tool, but i know January me was not ready for that.

**Comment by 5e-Cleric** [comment]
Related to this, today we noticed an interesting matter in GMB, its platform features like search and "new" and "features" displays have been victims to a spam attack:

![GMBinder being invaded by call girl ads](https://github.com/user-attachments/assets/10c86bea-ca77-429a-87bd-3ceedd21cbb4)
